app/services/versioned/publish_service.rb

```diff
@@ -54,66 +54,74 @@ def set_first_published_at
     end
 
     def publish_new_images
-      return unless current_edition.lead_image_revision
+      return unless current_edition.image_revisions.any?
 
-      # we currently only use the lead image
-      current_edition.lead_image_revision.asset_manager_variants.each do |variant|
-        raise "Expected variant to be on asset manager" if variant.absent?
+      current_edition.image_revisions.each do |image_revision|
+        image_revision.assets.each do |asset|
+          raise "Expected asset to be on asset manager" if asset.absent?
 
-        if variant.draft?
-          Versioned::AssetManagerService.new.publish(variant)
-          variant.file.live!
+          if asset.draft?
+            Versioned::AssetManagerService.new.publish(asset)
+            asset.live!
+          end
         end
       end
     end
 
     def retire_old_images
       return unless live_edition
 
+      current_revisions = current_edition.image_revisions
+      current_revisions_by_image_id = current_revisions.group_by(&:image_id)
+
       live_edition.image_revisions.each do |revision|
-        next if revision == current_edition.lead_image_revision
+        next if current_revisions.include?(revision)
```

Comment:
Do we need to specifically check the file revision here, otherwise won't we redirect when changing the alt text?